BrokerLock/BrokerLock.c

```diff
@@ -0,0 +1,39 @@
+/**@file
+ * This file is part of the TASTE Runtime Common.
+ *
+ * @copyright 2021 N7 Space Sp. z o.o.
+ *
+ * TASTE Runtime Common was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <BrokerLock.h>
+
+#include <rtems.h>
+
+extern rtems_id brokerLock;
+
+void
+Broker_acquire_lock()
+{
+    rtems_semaphore_obtain(brokerLock, RTEMS_WAIT, RTEMS_NO_TIMEOUT);
```

Comment:
Just a question -> you told me that for the UART driver there were spurious wakeups, right? Shouldn't we protect against them here (loop and == RTEMS_SUCCESFFUL?)

Reply:
ack